src/Appwrite/Platform/Tasks/Restore.php

```diff
@@ -0,0 +1,300 @@
+        shell_exec($cmd);
+
+        $stderr = shell_exec('tail -1 ' . $logfile);
+
+        if (!str_contains($stderr, 'completed OK!')) {
+            Console::error(date('Y-m-d H:i:s') . ' Prepare failed:' . $stderr);
+            Console::exit();
+        }
+    }
+
+    public function restore(string $target, bool $cloud, string $datadir)
+    {
+        $this->log('Restore start');
+
+        if (!file_exists($target)) {
+            Console::error('restore error directory not found: ' . $target);
+            Console::exit();
+        }
+
+        $logfile = $target . '/../log.txt';
+
+        $args = [
+            'xtrabackup',
+            $cloud ? '--move-back' : '--copy-back',
+            '--strict',
+            '--parallel=' . $this->processors,
+            '--target-dir=' . $target,
+            '--datadir=' . $datadir,
+            '2> ' . $logfile,
+        ];
+
+        $cmd = 'docker exec ' . $this->xtrabackupContainerId . ' ' . implode(' ', $args);
+        shell_exec($cmd);
+
+        $stderr = shell_exec('tail -1 ' . $logfile);
+
+        if (!str_contains($stderr, 'completed OK!')) {
+            Console::error(date('Y-m-d H:i:s') . ' Restore failed: ' . $stderr);
+            Console::exit();
+        }
+    }
+
+    public function checkEnvVariables(): void
+    {
+        foreach (
+            [
+                '_APP_CONNECTIONS_BACKUPS_STORAGE',
+                '_APP_CONNECTIONS_DB_REPLICAS',
+            ] as $env
+        ) {
+            if (empty(App::getEnv($env))) {
+                Console::error('Can\'t read ' . $env);
+                Console::exit();
+            }
+        }
+    }
+
+    public function log(string $message): void
+    {
+        if (!empty($message)) {
+            Console::log(date('Y-m-d H:i:s') . ' ' . $message);
+        }
+    }
+
+    public function setContainerId()
+    {
+        $stdout = '';
+        $stderr = '';
+        Console::execute('docker ps -f "name=xtrabackup" --format "{{.ID}}"', '', $stdout, $stderr);
+        if (!empty($stderr)) {
+            Console::error('Error setting container Id: ' . $stderr);
+            Console::exit();
+        }
+
+        $containerId = str_replace(PHP_EOL, '', $stdout);
+        if (empty($containerId)) {
+            Console::error('Xtrabackup Container ID not found');
+            Console::exit();
+        }
+
+        $this->xtrabackupContainerId = $containerId;
+    }
+
+    public function setProcessors()
+    {
+        $stdout = '';
+        $stderr = '';
+        Console::execute('docker exec ' . $this->xtrabackupContainerId . ' nproc', '', $stdout, $stderr);
+        if (!empty($stderr)) {
+            Console::error('Error setting processors: ' . $stderr);
+            Console::exit();
+        }
+
+        $processors = str_replace(PHP_EOL, '', $stdout);
+        $processors = intval($processors);
+
+        if ($processors === 0) {
+            Console::error('Set Processors Error');
```

Comment:
Let's set processors to 1 and return from here, shouldn't be a fatal error

Reply:
Done!